bookkeeper-server/src/main/java/org/apache/bookkeeper/client/BookKeeperAdmin.java

```diff
@@ -442,6 +449,69 @@ public SyncObject() {
         }
     }
 
+    public SortedMap<Long, LedgerMetadata> getLedgersContainBookies(Set<BookieSocketAddress> bookies)
+            throws InterruptedException, BKException {
+        final SyncObject sync = new SyncObject();
```

Comment:
I know this is kind of cherry pick.
Can you consider dropping this and use CompletableFuture. This legacy way of coding can be affected by spurious wake ups

Reply:
I would defer this to a separate task. BookKeeperAdmin needs a cleanup. I don't want to mix two approaches in the same file. how does that sound?

Reply:
sure, it is best to make one step at a time